core/windows/winmain.cpp

```diff
@@ -245,7 +249,66 @@ void UpdateInputState(u32 port)
 			DiscSwap();
 	}
 
+void UpdateController(u32 port)
+	{
+		XINPUT_STATE state;
+		
+		if (XInputGetState(port, &state) == 0)
+		{
+			WORD xbutton = state.Gamepad.wButtons;
+
+			if (xbutton & XINPUT_GAMEPAD_A) 
+				kcode[port] &= ~key_CONT_A;
+			if (xbutton & XINPUT_GAMEPAD_B)
+				kcode[port] &= ~key_CONT_B;
+			if (xbutton & XINPUT_GAMEPAD_Y)
+				kcode[port] &= ~key_CONT_Y;
+			if (xbutton & XINPUT_GAMEPAD_X)
+				kcode[port] &= ~key_CONT_X;
+
+			if (xbutton & XINPUT_GAMEPAD_START)
+				kcode[port] &= ~key_CONT_START;
+
+			if (xbutton & XINPUT_GAMEPAD_DPAD_UP)
+				kcode[port] &= ~key_CONT_DPAD_UP;
+			if (xbutton & XINPUT_GAMEPAD_DPAD_DOWN)
+				kcode[port] &= ~key_CONT_DPAD_DOWN;
+			if (xbutton & XINPUT_GAMEPAD_DPAD_LEFT)
+				kcode[port] &= ~key_CONT_DPAD_LEFT;
+			if (xbutton & XINPUT_GAMEPAD_DPAD_RIGHT)
+				kcode[port] &= ~key_CONT_DPAD_RIGHT;
+
+			lt[port] |= state.Gamepad.bLeftTrigger;
+			rt[port] |= state.Gamepad.bRightTrigger;
+
+			joyx[port] |=  state.Gamepad.sThumbLX / 257;
+			joyy[port] |= -state.Gamepad.sThumbLY / 257;
+		}
+	}
+
+void UpdateVibration(u32 port, u32 value)
+{
+		const u8 freq_l = 0x16;
```

Comment:
freq_l and freq_h ones? Yes, I suppose those don't even work as reference anymore.